ts/input/tex/ParseUtil.ts

```diff
@@ -33,668 +33,667 @@ import {entities} from '../../util/Entities.js';
 import {MmlMunderover} from '../../core/MmlTree/MmlNodes/munderover.js';
 
 
-namespace ParseUtil {
-
-  // TODO (VS): Combine some of this with lengths in util.
-  const emPerInch = 7.2;
-  const pxPerInch = 72;
-  // Note, the following are TeX CM font values.
-  const UNIT_CASES: {[key: string]: ((m: number) => number)}  = {
-    'em': m => m,
-    'ex': m => m * .43,
-    'pt': m => m / 10,                    // 10 pt to an em
-    'pc': m => m * 1.2,                   // 12 pt to a pc
-    'px': m => m * emPerInch / pxPerInch,
-    'in': m => m * emPerInch,
-    'cm': m => m * emPerInch / 2.54, // 2.54 cm to an inch
-    'mm': m => m * emPerInch / 25.4, // 10 mm to a cm
-    'mu': m => m / 18,
-  };
-  const num = '([-+]?([.,]\\d+|\\d+([.,]\\d*)?))';
-  const unit = '(pt|em|ex|mu|px|mm|cm|in|pc)';
-  const dimenEnd = RegExp('^\\s*' + num + '\\s*' + unit + '\\s*$');
-  const dimenRest = RegExp('^\\s*' + num + '\\s*' + unit + ' ?');
-
-
-  /**
-   * Matches for a dimension argument.
-   * @param {string} dim The argument.
-   * @param {boolean} rest Allow for trailing garbage in the dimension string.
-   * @return {[string, string, number]} The match result as (Anglosaxon) value,
-   *     unit name, length of matched string. The latter is interesting in the
-   *     case of trailing garbage.
-   */
-  export function matchDimen(
-    dim: string, rest: boolean = false): [string, string, number] {
-      let match = dim.match(rest ? dimenRest : dimenEnd);
-      return match ?
-        muReplace([match[1].replace(/,/, '.'), match[4], match[0].length]) :
-        [null, null, 0];
-  }
-
-
-  /**
-   * Transforms mu dimension to em if necessary.
-   * @param {[string, string, number]} [value, unit, length] The dimension triple.
-   * @return {[string, string, number]} [value, unit, length] The transformed triple.
-   */
-  function muReplace([value, unit, length]: [string, string, number]): [string, string, number] {
-    if (unit !== 'mu') {
-      return [value, unit, length];
-    }
-    let em = Em(UNIT_CASES[unit](parseFloat(value || '1')));
-    return [em.slice(0, -2), 'em', length];
+// TODO (VS): Combine some of this with lengths in util.
+const emPerInch = 7.2;
+const pxPerInch = 72;
+// Note, the following are TeX CM font values.
+export const UNIT_CASES: Map<string, number> = new Map([
+  ['em', 1],
+  ['ex', .43],
+  ['pt', 1 / 10],                // 10 pt to an em
+  ['pc', 1.2],                   // 12 pt to a pc
+  ['px', emPerInch / pxPerInch],
+  ['in', emPerInch],
+  ['cm', emPerInch / 2.54], // 2.54 cm to an inch
+  ['mm', emPerInch / 25.4], // 10 mm to a cm
+  ['mu', 1 / 18],
+]);
+
+const num = '([-+]?([.,]\\d+|\\d+([.,]\\d*)?))';
+const unit = () => `(${Array.from(UNIT_CASES.keys()).join('|')})`;
+const dimenEnd = () => RegExp('^\\s*' + num + '\\s*' + unit() + '\\s*$');
+const dimenRest = () => RegExp('^\\s*' + num + '\\s*' + unit() + ' ?');
```

Comment:
I'm a little concerned about the change of these, from an efficiency standpoint.  Every time a dimension is requests, you have to reconstruct the regular expressions, even when the units haven't changed.  That seems wasteful.

I know you are trying to allow units to be added, but wouldn't it be better to have an `addUnit()` function that, in addition to adding a new unit, also updates the regular expressions once, so they can be reused until a new unit is added?  (And perhaps a `removeUnit()` to balance it?)

Reply:
Added `addUnit`, `removeUnit`, and `getUnit`. The latter now allows us to always use `pt` if illegal unit is provided, similar to TeX behaviour.